File: dbv-nfc-ios/dbv-nfc-games/Models/Submission.swift
import Foundation

struct PlayerSubmissionRequest: Codable {
    let baseId: UUID
    let challengeId: UUID
    let answer: String
}

struct SubmissionResponse: Codable, Identifiable {
    let id: UUID
    let teamId: UUID
    let challengeId: UUID
    let baseId: UUID
    let answer: String
    let status: String
    let submittedAt: String
    let reviewedBy: UUID?
    let feedback: String?
}

File: web-admin/src/types/submission.ts
export type SubmissionStatus =
  | 'pending'
  | 'approved'
  | 'rejected'
  | 'correct'

export interface Submission {
  id: string
  teamId: string
  challengeId: string
  baseId: string
  answer: string
  fileUrl?: string
  fileUrls?: string[]
  status: SubmissionStatus
  submittedAt: string
  reviewedBy?: string
  feedback?: string
  points?: number
  completionContent?: string
}
